### PR TITLE
Simplify optimizer configs by hiding `OptimizerFactory` in `LitModular` (and `Adversary`)

### LINKED ISSUE #151: Wrap optimizer in mart.modules.Modular in mart.optim.OptimizerFactory
Shall we add this to `mart/models/modular.py` so that we can simplify the config in `mart/configs/optimizer`? That will make it easier to add other off-the-shelf optimizers from PyTorch.

_Originally posted by @mzweilin in https://github.com/IntelLabs/MART/pull/147#discussion_r1213437268_

### PR DESCRIPTION
**Pull request by mzweilin**
# What does this PR do?

Fixes #151

Depends on #147

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [x] `pytest`
- [x] `python -m mart experiment=CIFAR10_CNN_Adv trainer=gpu` achieves 70.36% accuracy.

## Before submitting

- [x] The title is **self-explanatory** and the description **concisely** explains the PR
- [x] My **PR does only one thing**, instead of bundling different changes together
- [x] I list all the **breaking changes** introduced by this pull request
- [x] I have commented my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have run pre-commit hooks with `pre-commit run -a` command without errors

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: mart/configs/optimizer/sgd.yaml [hunk 1]
```diff
@@ -1,9 +1,3 @@
...
-optimizer:
-  _target_: hydra.utils.get_method
-  path: torch.optim.SGD
+_target_: torch.optim.SGD
+_partial_: true
 lr: ???
-momentum: 0
-weight_decay: 0
-bias_decay: 0
-norm_decay: 0
```
Comment (dxoigmn) [inline mart/configs/optimizer/sgd.yaml:9]: I wouldn't change this given that it's often useful to set decay values in configuration.

**Review by dxoigmn — CHANGES_REQUESTED** [review]

File: mart/configs/optimizer/sgd.yaml (hunk 1 above)
Comment (mzweilin) [reply mart/configs/optimizer/sgd.yaml]: OK. I made the simplified config template a comment.

**Review by dxoigmn — APPROVED** [review]
LGTM!
